app/models/user.rb

```diff
@@ -26,8 +26,26 @@ class User < ApplicationRecord
   enum goal: { 'Gain Muscle' => 0, 'Lose Weight' => 1, 'Maintain Weight' => 2,
                'Increase Flexibility' => 3, 'Increase Stamina' => 4 }
 
-  def upcoming_events
-    events.where('date_time >= ?', Time.zone.now)
+  def upcoming_events(invitee_id = nil)
+    hosted = events.where('date_time >= ?', Time.zone.now)
+                   .order(date_time: :desc)
+    return hosted if invitee_id.nil?
+
+    invited = Event.where('date_time >= ?', Time.zone.now)
+                   .where(user_id: invitee_id)
+                   .order(date_time: :desc)
+    (hosted + invited).uniq
+  end
+
+  def past_events(invitee_id = nil)
+    hosted = events.where('date_time < ?', Time.zone.now)
+                   .order(date_time: :desc)
+    return hosted if invitee_id.nil?
+
+    invited = Event.where('date_time < ?', Time.zone.now)
+                   .where(user_id: invitee_id)
+                   .order(date_time: :desc)
+    (hosted + invited).uniq
   end
```

Comment:
I like what you did here.  For refactoring, I think we can abstract these methods out into the Event model as class methods because they are retrieving events.  Then in the User model, we can delegate these methods to the Event model.

Reply:
Refactored accordingly, thanks for the navigation support!!